### PR TITLE
Add query interface

### PR DESCRIPTION
**Pull request by lyppg**
do_check_key
do_get_match_last_index

### CODE REVIEW

File: src/infinistore.cpp [hunk 1]
```diff
@@ -655,6 +699,21 @@ void on_read(uv_stream_t *stream, ssize_t nread, const uv_buf_t *buf) {
...
+                        case OP_CHECK_EXIST:
+                            client->key_to_check.assign(client->recv_buffer,
+                                                        client->expected_bytes);
+                            handle_request(client);
+                            break;
+                        case OP_GET_MATCH_LAST_IDX:
+                            INFO("OP_GET_MATCH_LAST_IDX");
+                            if (!deserialize(client->recv_buffer, client->expected_bytes,
+                                             client->keys_meta)) {
+                                printf("failed to deserialize keys meta\n");
```
Comment (thesues) [inline src/infinistore.cpp:711]: no printf...

File: infinistore/lib.py [hunk 2]
```diff
@@ -393,3 +393,15 @@ def _verify(self, cache: torch.Tensor):
...
+
+    def check_exist(self, key: str):
+        ret = _infinistore.check_exist(self.conn, key)
+        if ret < 0:
+            raise Exception("Failed to check if this key exists")
+        return True if ret == 0 else False
+
+    def get_match_last_index(self, keys: List[str]):
+        ret = _infinistore.get_match_last_index(self.conn, keys)
+        if ret < 0:
```
Comment (thesues) [inline infinistore/lib.py:405]: always raise exception, we do not use return value for error

File: src/infinistore.cpp (hunk 1 above)
Comment (lyppg) [reply src/infinistore.cpp:711]: this is just copy/paste from previous code:) I will fix all printf

File: src/infinistore.cpp [hunk 3]
```diff
@@ -71,6 +71,9 @@ struct Client {
     rdma_conn_info_t remote_info;
     rdma_conn_info_t local_info;
 
+    std::string key_to_check;
```
Comment (thesues) [inline src/infinistore.cpp:74]: is it OK to put these two into STACK? because these could be deserilized from recv_buf?

Comment (lyppg) [reply src/infinistore.cpp:74]: I don't want to add another parameter in handle_request, we can do it when tcp proto is refined. 
or do you want me to process this request directly in on_read?

Comment (thesues) [reply src/infinistore.cpp:74]: we could do it later ;-)

File: src/infinistore.cpp (hunk 1 above)
Comment (lyppg) [reply src/infinistore.cpp]: fixed

File: infinistore/lib.py (hunk 2 above)
Comment (lyppg) [reply infinistore/lib.py:405]: fixed

**Pull request merged**